install_config/aggregate_logging.adoc

```diff
@@ -3847,3 +3850,230 @@ cluster. Edit your non-cluster logging service (for example, `logging-es`,
 ----
 $ oc patch svc/logging-es -p '{"spec":{"selector":{"component":"es","provider":"openshift"}}}'
 ----
+
+[[logging-troubleshooting]]
+== Troubleshooting
+
+Following is troubleshooting information for a number of commonly identified issues with cluster logging deployments:
+
+===  All Components
+
+*Deployment fails, RCs scaled to 0*
+
+When a deployment is performed, if it does not successfully bring up an
+instance before a ten-minute timeout, it will be considered failed and
+scaled down to zero instances. `oc get pods` will show a deployer pod
+with a non-zero exit code, and no deployed pods, e.g.:
+
+----
+NAME                           READY     STATUS             RESTARTS   AGE
+logging-es-2e7ut0iq-1-deploy   1/1       ExitCode:255       0          1m
+----
+
+(In this example, the deployer pod name for an Elasticsearch deployment is shown;
+this is from ReplicationController `logging-es-2e7ut0iq-1` which is a deployment
+of DeploymentConfig `logging-es-2e7ut0iq`.)
+
+Deployment failure can happen for a number of transitory reasons, such as
+the image pull taking too long, or nodes being unresponsive. Examine the
+deployer pod logs for possible reasons; but often you can redeploy:
+
+----
+$ oc deploy --latest logging-es-2e7ut0iq
+----
+
+Or you may be able to scale up the existing deployment:
+
+----
+$ oc scale --replicas=1 logging-es-2e7ut0iq-1
+----
+
+If the problem persists, examine pod, events, and systemd unit
+logs to determine the source of the problem.
+
+*Image pull fails*
+
+If you specify an `openshift_logging_image_prefix` that results in images being defined that don't exist,
+you will receive a corresponding error message:
+
+----
+NAME                     READY     STATUS                                                                                       RESTARTS   AGE
+logging-fluentd-1ub9k    0/1       Error: image registry.access.redhat.com:5000/openshift3logging-fluentd:latest not found      0          1m
+----
+
+In this example, for the intended image name
+`registry.access.redhat.com:5000/openshift3/logging-fluentd:latest`
+the `openshift_logging_image_prefix` needed a trailing `/`.
+
```

Comment:
@jcantrill What is the proper formatting for this parameter? (We don't list it in the docs)

`openshift_logging_image_prefix=registry.example.com/openshift3/`

Reply:
@ewolinetz can you advise?  it should be `<registry>/<namespace>/` right?

Reply:
I believe in this case we don't use a `namespace` because its not hosted on the cluster's image registry.

Reply:
@anpingli @ewolinetz @jcantrill 
The troubleshooting, I believe, is for 3.11 only. Would this be correct:
_If you use an image registry other than the default at `registry.redhat.io`, using `oreg_url` in the *_/etc/ansible/hosts_* file, you must supply a trailing `/` with that variable._

Reply:
@mburke5678  the oreg_url value is a template string [2].  No slash issue for oreg_url.   The customer shouldn't hit error [1] in v3.11.

In v3.11, there are two ways to change the logging image in playbook. ( openshift_logging_image_prefix is dropped )
1) set oreg_url, the openshift-ansible can get image_prefix from this value
2) Appoint image URL for some of the following variables
openshift_logging_fluentd_image=registry.example.com/openshift3/origin-logging-fluentd:v3.11.0
openshift_logging_elasticsearch_image=registry.example.com/openshift3/ose-logging-elasticsearch5:v3.11.0
openshift_logging_elasticsearch_proxy_image=registry.example.com/openshift3/oauth-proxy:v3.11.0
openshift_logging_kibana_image=registry.example.com/openshift3/openshift3/ose-logging-kibana5:v3.11.0
openshift_logging_kibana_proxy_image=registry.example.com/openshift3/oauth-proxy:v3.11.0
openshift_logging_curator_image=registry.example.com/openshift3/openshift3/ose-logging-curator5:v3.11.0

[1] 'Error: image registry.access.redhat.com:5000/openshift3logging-fluentd:latest' 
[2]  oreg_url=registry.access.redhat.com/openshift/ose-${component}:${version}